ts/fast-sync/index.ts

```diff
@@ -27,9 +29,17 @@ export class FastSyncSender {
         this.events.emit('prepared', { syncInfo })
         await channel.sendSyncInfo(syncInfo)
 
+        this.events.emit('progress', { progress: {
+                totalObjectsProcessed: this.totalObjectsProcessed,
+                lastObjectsProcessed: 0
+        } })
+
         for (const collection of this.options.collections) {
             for await (const objects of streamObjectBatches(this.options.storageManager, collection)) {
                 await channel.sendObjectBatch({ collection, objects })
+                this.events.emit('progress',{progress: {
+                    totalObjectsProcessed: this.totalObjectsProcessed+=objects.length,
```

Comment:
Generally, I try to avoid more complex combinations of statements and expression, and do the increment first in another line, then read it on another one.